packages/jupyter-ai/src/completions/plugin.ts

```diff
@@ -68,79 +88,100 @@ export const inlineCompletionProvider: JupyterFrontEndPlugin<void> = {
+      newJaiIcpSettings: Partial<JaiInlineProvider.ISettings>
+    ) {
+      const icSettings = await getIcSettings();
+      const oldUserIcpSettings = icSettings.user.providers;
+      const newUserIcpSettings = {
+        ...oldUserIcpSettings,
+        [JaiInlineProvider.ID]: {
+          ...oldUserIcpSettings?.[JaiInlineProvider.ID],
+          ...newJaiIcpSettings
         }
-        const providers = Object.assign({}, settings.user.providers) as any;
-        const ourSettings = {
-          ...JupyterAIInlineProvider.DEFAULT_SETTINGS,
-          ...providers[provider.identifier]
-        };
-        const wasEnabled = ourSettings['enabled'];
-        providers[provider.identifier]['enabled'] = !wasEnabled;
-        settings.set('providers', providers);
+      };
+      icSettings.set('providers', newUserIcpSettings);
+    }
+
+    app.commands.addCommand(CommandIDs.toggleCompletions, {
+      execute: async () => {
+        const jaiIcpSettings = await getJaiIcpSettings();
+        updateJaiIcpSettings({
+          enabled: !jaiIcpSettings.enabled
+        });
       },
-      label: 'Enable Jupyternaut Completions',
+      label: 'Enable completions by Jupyternaut',
       isToggled: () => {
         return provider.isEnabled();
       }
     });
 
     app.commands.addCommand(CommandIDs.toggleLanguageCompletions, {
-      execute: () => {
+      execute: async () => {
+        const jaiIcpSettings = await getJaiIcpSettings();
         const language = findCurrentLanguage();
-        if (!settings || !language) {
+        if (!language) {
           return;
         }
-        const providers = Object.assign({}, settings.user.providers) as any;
-        const ourSettings = {
-          ...JupyterAIInlineProvider.DEFAULT_SETTINGS,
-          ...providers[provider.identifier]
-        };
-        const wasDisabled = ourSettings['disabledLanguages'].includes(
-          language.name
-        );
-        const disabledList: string[] =
-          providers[provider.identifier]['disabledLanguages'];
-        if (wasDisabled) {
-          disabledList.filter(name => name !== language.name);
-        } else {
-          disabledList.push(language.name);
-        }
-        settings.set('providers', providers);
+
+        const disabledLanguages = [...jaiIcpSettings.disabledLanguages];
+        const newDisabledLanguages = disabledLanguages.includes(language.name)
+          ? disabledLanguages.filter(l => l !== language.name)
+          : disabledLanguages.concat(language.name);
+
+        updateJaiIcpSettings({
+          disabledLanguages: newDisabledLanguages
+        });
       },
       label: () => {
         const language = findCurrentLanguage();
         return language
-          ? `Enable Completions in ${displayName(language)}`
-          : 'Enable Completions for Language of Current Editor';
+          ? `Disable completions in ${displayName(language)}`
+          : 'Disable completions in <language> files';
```

Comment:
Ah, this switch gave me confusion as a user. All other UI elements in JupyterLab use positive mode.